src/main/scala/io/iohk/ethereum/vm/Memory.scala

```diff
@@ -70,6 +70,21 @@ class Memory private(private val underlying: ByteString) {
       (newUnderlying.slice(start, end), new Memory(newUnderlying))
     }
 
+  /** This function will expand the Memory size as if storing data given the `offset` and `size`.
+    * If the memory is already initialised at that region it will not be modified, otherwise it will be filled with
+    * zeroes.
+    * This is required to satisfy memory expansion semantics for *CALL* opcodes.
+    */
+  def expand(offset: UInt256, size: UInt256): Memory = {
+    val totalSize = (offset + size).toInt
+    if (this.size >= totalSize || size.isZero)
```

Comment:
@rtkaczyk found this `size.isZero` was missing. Based on YP EIP-150 REVISION above eq 222

> Referencing a zero length range (e.g. by attempting to pass it as the input range to a CALL) does not require memory to be extended to the beginning of the range.